New CLI examples for RDS Performance Insights

daniil-millwood opened the pull request:
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

elysahall reviewed (changes requested):
Thank you for your submission. Please make the requested changes, squash your commits, and push to the PR.

daniil-millwood commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+
+    {
+        "Dimensions":[
+        {
```
Okay. Edited file to add these indents

elysahall reviewed (approved):
LGTM

ashovlin commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+        --service-type RDS \
+        --identifier db-10BCD2EFGHIJ3KL4M5NO6PQRS5 \
+        --group db.sql \
+        --group-identifier my-sql-id \
```
Will `--group-identifier` always be something not human readable? It was something like `5604BCCE604DCF7A536C06C84E5E0B0BD50E3802` when I tested this. 

If so, can we either show a realistic value, or otherwise indicate that this is a placeholder?

daniil-millwood replied:
Yes, changing the value to `example-sql-id`

ashovlin commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+        ]
+    }
+
+For more information about dimensions in Performance Insights, see `Database load <https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.Overview.ActiveSessions.html>`__ in the *Amazon RDS User Guide* and `Database load <https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_PerfInsights.Overview.ActiveSessions.html>`__ in the *Amazon Aurora User Guide*.
```
Is the database load link still relevant here? The example is for `db.sql`, not `db.load`.

daniil-millwood replied:
Yes, the page is relevant because it describes Top SQL and dimensions.

ashovlin commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+
+**Example 2: To get details for a specified dimension group for a DB instance (Windows)**
+
+The following ``get-dimension-key-details`` example retrieves the full text of a SQL statement for DB instance ``db-10BCD2EFGHIJ3KL4M5NO6PQRS5``. The ``--group`` is ``db.sql``, and the ``--group-identifier`` is ``db.sql.id``. In this example, ``my-sql-id`` represents a SQL ID retrieved by using the ``get-resource-metrics`` or ``describe-dimension-keys`` operations. ::
```
I think it'd be complementary to add an example to `get-resource-metrics` or `describe-dimension-keys` to show how to retrieve `my-sql-id`. It wasn't obvious to me and took a while to get there.

daniil-millwood replied:
Adding second example for `describe-dimension-keys.rst`

daniil-millwood commented on awscli/examples/pi/delete-performance-analysis-report.rst:
```diff
@@ -0,0 +1,13 @@
+**To delete a performance analysis report**
+
+The following ``delete-perforamnce-analysis-report`` example deletes the performance analysis report with the report ID ``report-0d99cc91c4422ee61``. ::
```
Fixed.

daniil-millwood commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+Output::
+
+    {
+        "Dimensions":[
```
Fixed.

elysahall commented on awscli/examples/pi/delete-performance-analysis-report.rst:
```diff
@@ -0,0 +1,13 @@
+        --service-type RDS \
+        --identifier db-loadtest-0 \
+        --analysis-report-id report-0d99cc91c4422ee61 \
+        --region us-west-2
```
We usually don't specify regions if it's ``us-west-2``, as we tend to default examples to using ``us-west-2``.

daniil-millwood replied:
Removing `--region us-west-2` from all files.

daniil-millwood commented on awscli/examples/pi/delete-performance-analysis-report.rst:
```diff
@@ -0,0 +1,13 @@
+
+    aws pi delete-performance-analysis-report \
+        --service-type RDS \
+        --identifier db-loadtest-0 \
```
Yes, please change the --identifier value to db-ABCDEFGHIJKLMNOPQRSTU1VWZ.

daniil-millwood replied:
All caps for the dummy resource is a bit loud and doesn't follow the AWS style guide (I know that example is used in other AWS docs). I'm updating all instances with db-abcdefg123456789.

daniil-millwood commented on awscli/examples/pi/list-performance-analysis-reports.rst:
```diff
@@ -0,0 +1,45 @@
+            {
+                "Status": "Succeeded",
+                "EndTime": 1680587086.584,
+                "CreationTime": 1680587087.139,
```
Updating to `CreateTime`

daniil-millwood commented on awscli/examples/pi/get-resource-metadata.rst:
```diff
@@ -0,0 +1,20 @@
+**To get resource metadata for a database**
+
+The following ``get-resource-metadata`` example gets the resource metadata for the database ``db-loadtest-0`` with the report ID ``report-0d99cc91c4422ee61``. The response shows that SQL digest statistics are enabled. ::
```
Removing.

daniil-millwood commented on awscli/examples/pi/get-dimension-key-details.rst:
```diff
@@ -0,0 +1,51 @@
+
+For more information about dimensions in Performance Insights, see `Database load <https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.Overview.ActiveSessions.html>`__ in the *Amazon RDS User Guide* and `Database load <https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_PerfInsights.Overview.ActiveSessions.html>`__ in the *Amazon Aurora User Guide*.
+
+**Example 2: To get details for a specified dimension group for a DB instance (Windows)**
```
Removing the Windows example.

daniil-millwood commented:
@ashovlin I addressed your feedback from last week and forced the changes to the same commit. Could you please review the updates? I have not run the Travis CI tests. I don't have permissions like Elysa to do this. Please run tests and continue with your review.

Thanks,
Daniil
